src/main/resources/templates/pagination.html

```diff
@@ -0,0 +1,20 @@
+<th:block th:fragment="pagination(pageable, page)">
+    <nav aria-label="Page navigation example" th:if="${page?.totalPages > 1}">
+        <ul class="pagination justify-content-end">
+            <li class="page-item" th:classappend="${page.first} ? 'disabled'">
+                <a class="page-link"
+                   th:href="@{/company(page=${page.number - 1}, size=${page.size})}"
+                   tabindex="-1" aria-disabled="true">Previous</a>
+            </li>
+            <li class="page-item" th:classappend="${currentPage == page.number} ? 'active'"
+                th:each="currentPage : ${#numbers.sequence(0, page.totalPages - 1)}">
+                <a class="page-link" th:href="@{/company(page=${currentPage}, size=${page.size})}"
+                   th:text="${currentPage + 1}"></a>
+            </li>
+            <li class="page-item" th:classappend="${page.last} ? 'disabled'">
+                <a class="page-link"
+                   th:href="@{/company(page=${page.number + 1}, size=${page.size})}">Next</a>
```

Comment:
There should not be anything specific to **Company** here. In case needed, pass it through the parameter.